src/Workleap.Extensions.OpenAPI/TypedResult/ExtractSchemaTypeResultFilter.cs

```diff
@@ -123,27 +123,21 @@ internal static IEnumerable<ResponseMetadata> GetResponsesMetadata(Type returnTy
     // Initialize an instance of the result type to get the response metadata and return null if it's not possible
     private static ResponseMetadata? ExtractMetadataFromTypedResult(Type resultType)
     {
+        var typeString = $"{resultType.Namespace}.{resultType.Name}";
         // For type like Ok, BadRequest, NotFound
         if (!resultType.GenericTypeArguments.Any())
         {
-            var constructor = resultType.GetConstructor(BindingFlags.NonPublic | BindingFlags.Instance, null, Array.Empty<Type>(), null);
-            if (constructor != null)
+            if (HttpResultsStatusCodeTypeHelpers.HttpResultTypeToStatusCodes.TryGetValue(typeString, out var statusCode))
```

Comment:
We could hide the implementation of the dictionary and the concat logic inside a private method: ExtractStatusCodeFromType()